components/alert/alert-toast.js

```diff
@@ -257,29 +272,50 @@ class AlertToast extends LitElement {
 		clearTimeout(this._setTimeoutId);
 	}
 
-	_handleAlertClose(e) {
-		if (e?.target === this || !this.open) return;
+	_handleButtonPress(e) {
+		e.stopPropagation();
+		this.dispatchEvent(new CustomEvent('d2l-alert-toast-button-press'));
+	}
+
+	_handlePageResize(e) {
+		this._smallWidth = e.matches;
+	}
 
-		const containerBottom = parseFloat(getComputedStyle(this._innerContainer).getPropertyValue('bottom'));
-		const closingContainerBottom = e.detail.bottom;
-		if (closingContainerBottom > containerBottom) return; // closing alert is above this alert, no need to adjust bottom spacing
+	_handleResize() {
+		const newHeight = this._innerContainer.offsetHeight;
+		const oldHeight = this._height;
+		this._height = newHeight;
+		if (newHeight === oldHeight || newHeight === 0) return; // do not run if height has not changed or if closed
 
-		this._bottomHeight -= e.detail.height;
-		this._bottomMargin -= TOAST_SPACING;
-		if (!reduceMotion) this._state = states.SLIDING;
+		const bottom = parseFloat(getComputedStyle(this._innerContainer).getPropertyValue('bottom'));
+		const opening = oldHeight === 0;
+
+		this.dispatchEvent(new CustomEvent(
+			'd2l-alert-toast-resize', {
+				bubbles: true,
+				composed: false,
+				detail: { bottom, heightDifference: (oldHeight - newHeight), opening, closing: false }
+			}
+		));
 	}
 
-	_handleAlertOpen(e) {
+	_handleSiblingResize(e) {
 		if (e?.target === this || !this.open) return;
 
-		this._bottomHeight += e.detail.height;
-		this._bottomMargin += TOAST_SPACING;
-		if (!reduceMotion) this._state = states.SLIDING;
-	}
+		if (!e.detail.opening) {
+			const containerBottom = parseFloat(getComputedStyle(this._innerContainer).getPropertyValue('bottom'));
+			const siblingContainerBottom = e.detail.bottom;
+			if (siblingContainerBottom > containerBottom) return; // resized alert is above this alert, no need to adjust bottom spacing
+		}
 
-	_handleButtonPress(e) {
-		e.stopPropagation();
-		this.dispatchEvent(new CustomEvent('d2l-alert-toast-button-press'));
+		this._bottomHeight -= e.detail.heightDifference;
```

Comment:
That is correct. I'll think on it more and see if I can simplify it a bit

Reply:
Maybe it could be a little clearer if we renamed `_bottomHeight` -> `_totalSiblingHeightBelow` or something along those lines? Something like that would also make it a little clearer that it doesn't include the space between them. And maybe reverse the difference calc so that `e.detail.heightDifference` is positive when the siblings are growing in height... unless of course there is another reason for it to be negative. 🤷‍♂️